editor/src/components/inspector/flex-section.tsx

```diff
@@ -1112,3 +1099,99 @@ const useGridExpressionInputFocused = () => {
   const onBlur = React.useCallback(() => setFocused(false), [])
   return { focused, onFocus, onBlur }
 }
+
+function parseGridDimensionInput(
+  value: UnknownOrEmptyInput<CSSNumber | CSSKeyword<ValidGridDimensionKeyword>>,
+  currentValue: GridDimension | null,
+) {
+  if (isCSSNumber(value)) {
+    const maybeUnit =
+      currentValue != null && isGridCSSNumber(currentValue) ? currentValue.value.unit : null
+    return gridCSSNumber(
+      cssNumber(value.value, value.unit ?? maybeUnit),
+      currentValue?.areaName ?? null,
+    )
+  } else if (isCSSKeyword(value)) {
+    return gridCSSKeyword(value, currentValue?.areaName ?? null)
+  } else if (isEmptyInputValue(value)) {
+    return gridCSSKeyword(cssKeyword('auto'), currentValue?.areaName ?? null)
+  } else {
+    return null
+  }
+}
+
+const AutoColsOrRowsControl = React.memo(
+  (props: { axis: 'column' | 'row'; grid: ElementInstanceMetadata }) => {
+    const value = React.useMemo(() => {
+      const template = props.grid.specialSizeMeasurements.containerGridPropertiesFromProps
+      const data = props.axis === 'column' ? template.gridAutoColumns : template.gridAutoRows
+      if (data?.type !== 'DIMENSIONS') {
+        return null
+      }
+      return data.dimensions[0]
+    }, [props.grid, props.axis])
+
+    const dispatch = useDispatch()
+
+    const onUpdateDimension = React.useCallback(
+      (newDimension: GridDimension) => {
+        dispatch([
+          applyCommandsAction([
+            setProperty(
+              'always',
+              props.grid.elementPath,
+              PP.create('style', props.axis === 'column' ? 'gridAutoColumns' : 'gridAutoRows'),
+              printArrayGridDimensions([newDimension]),
+            ),
+          ]),
+        ])
+      },
+      [props.grid, props.axis, dispatch],
+    )
+
+    const onUpdateAutoValue = React.useCallback(
```

Comment:
small naming thing - this can be called on any number/keyword, correct? (not just "auto" value)

Reply:
renamed